pkg/store/errors.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"errors"
+	"github.com/jackc/pgconn"
+	"github.com/mattn/go-sqlite3"
+)
+
+func IsUniqueConstraintErr(err error) bool {
+	if dbErr, ok := castDbErr(err); ok {
+		return dbErr.IsUniqueConstraintErr()
+	}
+	return false
+}
+
+func castDbErr(err error) (dbErr, bool) {
+	castSqliteErr := &sqlite3.Error{}
+	if errors.As(err, castSqliteErr) {
+		return &sqliteErr{err: castSqliteErr}, true
+	}
+	postgresErr := &pgconn.PgError{}
+	if errors.As(err, &postgresErr) {
+		return &pgErr{err: postgresErr}, true
+	}
+	return nil, false
+}
+
+type dbErr interface {
+	IsUniqueConstraintErr() bool
+}
+
+type pgErr struct {
+	err *pgconn.PgError
+}
+
+func (s *pgErr) IsErrCode(code string) bool {
+	return s.err.Code == code
+}
+
+func (s *pgErr) IsUniqueConstraintErr() bool {
+	return s.IsErrCode("23505")
```

Comment:
Where does this number come from?

Reply:
This is a documented PostgreSQL error code. 
https://www.postgresql.org/docs/13/errcodes-appendix.html

Reply:
Gonna add this link to the file

Reply:
May be worth making it a variable

Reply:
Just did ;)